### PR TITLE
Add option to modify individual jobs on application forms

### PR DESCRIPTION
**Pull request by vassyz**
This commit adds a "Change" link to the work history section of an application form. This link takes the support user to a new page where they can edit the details of a single job. It reuses the candidate interface form for editing jobs, but with a different controller and view.

 [Support: Edit "work history" functionality](https://trello.com/c/G87nTULF/826-support-edit-work-history-functionality)

### Adds a change button:
Screenshot 2023-10-20 at 13 36 42

### That takes you to the edit job page:
Screenshot 2023-10-20 at 13 36 11

### CODE REVIEW

**Comment by vassyz** [comment]
[Support: Edit "work history" functionality](https://trello.com/c/G87nTULF/826-support-edit-work-history-functionality)

File: spec/system/support_interface/change_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.feature 'Change job' do
+  include DfESignInHelpers
+
+  scenario 'Change the english GCSE on an application form', :with_audited do
```
Comment (tomas-stefano) [inline spec/system/support_interface/change_job_spec.rb:6]:
```suggestion
  scenario 'Change the work history on an application form', :with_audited do
```

**Review by tomas-stefano — APPROVED** [review]
All good 🚀 . Will we have a Add work history to support too?

File: spec/system/support_interface/change_job_spec.rb (hunk 1 above)
Comment (vassyz) [reply spec/system/support_interface/change_job_spec.rb:6]: Woopps I copied the first 5 lines of your test 😆

**Comment by vassyz** [comment]
> All good 🚀 . Will we have a Add work history to support too?

Yes, that's coming soon CC @JR-G

**Comment by JR-G** [comment]
> > All good 🚀 . Will we have a Add work history to support too?
> 
> Yes, that's coming soon CC @JR-G

_Coming soon to a support console near you_

**Pull request merged**